multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1261,6 +1261,7 @@ def _fit(
             monitor=task.validation_metric_name,
             patience=config.optimization.patience,
             mode=minmax_mode,
+            stopping_threshold=1 - 1e-6 if minmax_mode == MAX else 1e-6,
```

Comment:
Are we assuming all positive-is-better metrics have an optimal score of 1? I guess in general this is true, but it is a pretty flimsy assumption especially if a user can provide a custom metric. Ditto for assuming optimal score of 0 for negative-is-better metrics.

These situations are handled in my `autogluon.core.metrics` logic, where each metric has a defined optimum value.

Reply:
Beyond the above point, the fact we need an `if` here seems like unnecessary code bloat and increases risk of bugs. It would be better if the optimum value / minmax_mode was simply provided by the metric itself, rather than hard-coded into this line.

Reply:
for example:

`stopping_threshold = metric.optimum - metric.sign * epsilon`
which translates to:
`stopping_threshold = 1 - (1 * 1e-6)`

I think this would be a lot cleaner.

Both of these things are provided in `autogluon.core.metrics`.

```python3
from autogluon.core.metrics import roc_auc

epsilon = 1e-6
stopping_threshold = roc_auc.optimum - roc_auc._sign * epsilon

print(stopping_threshold)
```
```
0.999999
```

Reply:
Good suggestion. Thanks.